src/servant/create_event_entity.h

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+void CreateEventEntity(Entity* entityParent, s32 entityId, s32 params) {
+    Entity* entity;
+    s32 i;
+
+    for (i = 0; i < 3; i++) {
+        entity = &g_Entities[5 + i];
+        if (!entity->entityId) {
+            break;
+        }
+    }
+
+    if (!entity->entityId) {
+        DestroyEntity(entity);
+        entity->entityId = entityId;
+        entity->zPriority = entityParent->zPriority;
+        entity->facingLeft = entityParent->facingLeft;
+        entity->flags = FLAG_KEEP_ALIVE_OFFCAMERA;
+        entity->posX.val = entityParent->posX.val;
+        entity->posY.val = entityParent->posY.val;
+        // Not necessarily making batFamBlueTrail here, but
+        // that's an Ext that works. Just needs parent at 0x8C.
+        entity->ext.batFamBlueTrail.parent = entityParent;
```

Comment:
Should I change this now it's shared between entities? Perhaps a `ext.servant` extension?

Reply:
I think we can adjust this as we decompile more of the familiars and see what's really in common, this is fine for now.